guides/getting-started.md

```diff
@@ -0,0 +1,1138 @@
+ create  .Resources.NATSecurityGroup.Properties.SecurityGroupEgress.0.CidrIp: 0.0.0.0/0
+ create  .Resources.NATSecurityGroup.Properties.SecurityGroupEgress.1.IpProtocol: 6
+ create  .Resources.NATSecurityGroup.Properties.SecurityGroupEgress.1.FromPort: 80
+ create  .Resources.NATSecurityGroup.Properties.SecurityGroupEgress.1.ToPort: 80
+ create  .Resources.NATSecurityGroup.Properties.SecurityGroupEgress.1.CidrIp: 0.0.0.0/0
+```
+
+It's exactly what we expect. Looks like it's just the ingress and egress rules,
+so we can converge the template and apply the changes.
+
+```text
+$> convection converge
+
+converge  Stack vpc
+update_in_progress  convection-demo-vpc: (AWS::CloudFormation::Stack) User Initiated
+update_in_progress  NATSecurityGroup: (AWS::EC2::SecurityGroup)
+update_complete  NATSecurityGroup: (AWS::EC2::SecurityGroup)
+update_complete_cleanup_in_progress  convection-demo-vpc: (AWS::CloudFormation::Stack)
+update_complete  convection-demo-vpc: (AWS::CloudFormation::Stack)
+```
+
+## Create the NAT router ##
+
+Now that we've got our security group, we need to set up an EC2 instance that
+will function as a NAT router. We'll be using one of Amazon's pre-built NAT
+images since we don't need anything custom. The
+[AWS::EC2::Instance][cf-ec2] resource handles creating our
+router instance from the given AMI. Since our router provides internet access,
+it needs to be in the public subnet. We also need to disable source/destination
+checking so it can perform network address translation. Finally, we'll make
+sure the instance is in the security group we just created.
+
+Update your Cloudfile to look like the one below.
+
+```ruby
+require 'convection'
+
+class Model::Template::Resource::EC2Subnet
+  property :public_ips, 'MapPublicIpOnLaunch', :default => false
+end
+
+name 'convection-demo'
+region 'us-east-1'
+
+vpc = template do
+  description 'VPC with Public and Private Subnets (NAT)'
+
+  ec2_vpc 'DemoVPC' do
+    network '10.10.10.0/23'
+    tag 'Name', "#{stack.cloud}-#{stack.name}"
+  end
+
+  ec2_subnet 'PrivateSubnet' do
+    network '10.10.10.0/24'
+    tag 'Name', "#{stack.cloud}-#{stack.name}-private"
+    vpc fn_ref('DemoVPC')
+  end
+
+  ec2_subnet 'PublicSubnet' do
+    network '10.10.11.0/24'
+    tag 'Name', "#{stack.cloud}-#{stack.name}-public"
+    vpc fn_ref('DemoVPC')
+    public_ips true
+  end
+
+  ec2_security_group 'NATSecurityGroup' do
+    description 'NAT access for private subnet'
+    vpc fn_ref('DemoVPC')
+    tag 'Name', "#{stack.cloud}-#{stack.name}-nat-security-group"
+    ingress_rule :tcp, 443 do
+      source '10.10.10.0/24'
+    end
+    ingress_rule :tcp, 80 do
+      source '10.10.10.0/24'
+    end
+    egress_rule :tcp, 443 do
+      source '0.0.0.0/0'
+    end
+    egress_rule :tcp, 80 do
+      source '0.0.0.0/0'
+    end
+  end
+
+  ec2_instance 'NATInstance' do
+    tag 'Name', "#{stack.cloud}-#{stack.name}-nat"
+    image_id 'ami-c02b04a8'
```

Comment:
How often does this AMI id change? Is someone going to have to update it every time Amazon releases a NAT AMI update?